extensions/amp-animation/0.1/css-expr-ast.js

```diff
@@ -0,0 +1,250 @@
+      set.array_.push(otherNodeOrSet);
+    }
+    return set;
+  }
+}
+
+
+/**
+ * A CSS expression that's simply passed through from the original expression.
+ * Used for `url()`, colors, etc.
+ */
+export class CssPassthroughNode extends CssNode {
+  /** @param {string} css */
+  constructor(css) {
+    super();
+    /** @const @private {string} */
+    this.css_ = css;
+  }
+}
+
+
+/**
+ * A CSS number: `100`, `1e2`, `1e-2`, `0.5`, etc.
+ */
+export class CssNumberNode extends CssNode {
+  /** @param {number} num */
+  constructor(num) {
+    super();
+    /** @const @private {number} */
+    this.num_ = num;
+  }
+}
+
+
+/**
+ * A CSS percent value: `100%`, `0.5%`, etc.
+ */
+export class CssPercentNode extends CssNode {
+  /** @param {number} num */
+  constructor(num) {
+    super();
+    /** @const @private {number} */
+    this.num_ = num;
+  }
+}
+
+
+/**
+ * A CSS length value: `100px`, `80vw`, etc.
+ */
+export class CssLengthNode extends CssNode {
+  /**
+   * @param {number} num
+   * @param {string} units
+   */
+  constructor(num, units) {
+    super();
+    /** @const @private {number} */
+    this.num_ = num;
+    /** @const @private {string} */
+    this.units_ = units;
+  }
+}
+
+
+/**
+ * A CSS angle value: `45deg`, `0.5rad`, etc.
+ */
+export class CssAngleNode extends CssNode {
+  /**
+   * @param {number} num
+   * @param {string} units
+   */
+  constructor(num, units) {
+    super();
+    /** @const @private {number} */
+    this.num_ = num;
+    /** @const @private {string} */
+    this.units_ = units;
+  }
+}
+
+
+/**
+ * A CSS time value: `1s`, `600ms`.
+ */
+export class CssTimeNode extends CssNode {
+  /**
+   * @param {number} num
+   * @param {string} units
+   */
+  constructor(num, units) {
+    super();
+    /** @const @private {number} */
+    this.num_ = num;
+    /** @const @private {string} */
+    this.units_ = units;
+  }
+}
+
+
+/**
+ * A CSS generic function: `rgb(1, 1, 1)`, `translateX(300px)`, etc.
+ */
+export class CssFuncNode extends CssNode {
+  /**
+   * @param {string} name
+   * @param {!Array<!CssNode>} args
+   */
+  constructor(name, args) {
+    super();
+    /** @const @private {string} */
+    this.name_ = endsWith(name, '(') ?
+        name.substring(0, name.length - 1) : name;
+    /** @const @private {!Array<!CssNode>} */
+    this.args_ = args;
+  }
+}
+
+
+/**
+ * A CSS `var()` expression: `var(--name)`, `var(--name, 100px)`, etc.
+ * See https://www.w3.org/TR/css-variables/.
+ */
+export class CssVarNode extends CssNode {
+  /**
+   * @param {string} varName
+   * @param {!CssNode=} opt_def
+   */
+  constructor(varName, opt_def) {
+    super();
+    /** @const @private {string} */
+    this.varName_ = varName;
+    /** @const @private {?CssNode} */
+    this.def_ = opt_def || null;
+  }
+}
+
+
+/**
+ * A CSS `calc()` expression: `calc(100px)`, `calc(80vw - 30em)`, etc.
+ * See https://drafts.csswg.org/css-values-3/#calc-notation.
+ */
+export class CssCalcNode extends CssNode {
```

Comment:
Technically no. `calc()` and `var()` are expressions - they resolve in actual values. While `translate()`, `rotate()` and such are actual CSS functions. The formats within are very different, e.g. `+` and `-` are only possible in `calc()` and no `,` args. Implementation-wise they will also differ a lot with `calc()` and `var()` actually resolving to final values, e.g. `100px`.